### PR TITLE
[WIP] Adding dialogue state tracking TRADE

### PR DESCRIPTION
**Pull request by VahidooX**
Adding TRADE (a dialogue state tracking model) to NeMo. 

TRADE is introduced in the following paper:
"Transferable Multi-Domain State Generator for Task-Oriented Dialogue Systems", Chien-Sheng Wu, Andrea Madotto, Ehsan Hosseini-Asl, Caiming Xiong, Richard Socher and Pascale Fung. ACL 2019.

The model is adopted from: 
https://github.com/jasonwu0731/trade-dst

### CODE REVIEW

File: examples/nlp/dialogue_state_tracking_trade.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+    data_size = len(data_layer)
+    logging.info(f'The length of data layer is {data_size}')
+
+    if data_size < batch_size:
+        logging.warning("Batch_size is larger than the dataset size")
+        logging.warning("Reducing batch_size to dataset size")
+        batch_size = data_size
+
+    steps_per_epoch = math.ceil(data_size / (batch_size * num_gpus))
```
Comment (tkornuta-nvidia) [inline examples/nlp/dialogue_state_tracking_trade.py]: I saw that formula several times, not sure whether it is totally correct - are we skipping the last batch if it doesn't have "enough samples"?

Comment (VahidooX) [reply examples/nlp/dialogue_state_tracking_trade.py]: This formula is supposed to round it up, so if last batch is smaller, then it is still counted as an step.

File: examples/nlp/scripts/multiwoz/process_multiwoz.py [hunk 2]
```diff
@@ -0,0 +1,398 @@
+# =============================================================================
```
Comment (tkornuta-nvidia) [inline examples/nlp/scripts/multiwoz/process_multiwoz.py]: I think scripts should start with a shebang:

#!/usr/bin/python

Comment (VahidooX) [reply examples/nlp/scripts/multiwoz/process_multiwoz.py]: Added to the file.

**Pull request merged**